test/unit/dimensions.js

```diff
@@ -646,7 +646,9 @@ QUnit.test( "width/height on a table row with phantom borders (gh-3698)", functi
 	} );
 } );
 
-QUnit.test( "interaction with scrollbars (gh-3589)", function( assert ) {
+// Because we fallback to offsetWidth/offsetHeight,
+// IE9-11 does not report fractional values
+QUnit[ jQuery.support.boxSizingReliable() ? "test" : "skip" ]( "interaction with scrollbars (gh-3589)", function( assert ) {
```

Comment:
These tests only ensure that fractional values are maintained. Is there another issue your concerned about or do you think fractional values are more important than I do?

Reply:
The tests are not for ensuring preservation of fractional values, but rather for verifying that `.{inner,outer}{Width,Height}` work correctly with scrollable elements as both getters and setters per #3589... doing so in all cases just happens to be tricky because the [fix](https://github.com/jquery/jquery/commit/80f1c8239e9d5f793f3e54e0cb8d7bd0747e4856#diff-414c8f59bff0d1b63680b64763d8c529R134) required arithmetic mixing possibly fractional CSS values with always-integer offsetWidth/offsetHeight, which is why that test block includes fractional CSS values—but note that it is explicitly uses `% 1` to work even on browsers that don't support them.

These tests are complex but IMO necessary, and they're working now. Any change that breaks them makes me uncomfortable, and any change that bypasses them is just begging for trouble.

Reply:
Let's run the tests in all environments but force them to be integer-only in IE like so:
```patch
-		fraction = jQuery( "<div style='width:4.5px;'/>" ).appendTo( parent ).width() % 1,
+		fraction = jQuery.support.boxSizingReliable() ?
+			jQuery( "<div style='width:4.5px;'/>" ).appendTo( parent ).width() % 1 :
+			0,
```

Reply:
Thanks for the explanation. I gave up on these tests too quickly then.

Reply:
Your suggestion works.

Reply:
Fixed IE9 by accounting for the element shrinking at the test level rather than trying to adjust the width. Our code compensates for scrollbars + border-box + position: absolute in IE9 only and it's like it was trying to compensate twice.